lib/network_interface/worker.ex

```diff
@@ -125,8 +126,37 @@ defmodule Nerves.NetworkInterface.Worker do
     { :ok, %Nerves.NetworkInterface.Worker{port: port} }
   end
 
+  #Returns intersection of lists a and b
+  defp intersect(a, b), do: a -- (a -- b)
```

Comment:
Do we need to remove duplicates too? This function doesn't understand duplicated.

The idiomatic way is `MapSet.intersection(MapSet.new(a), MapSet.new(b)) |> MapSet.to_list()`. I don't know if it is faster or better on memory. I thought you might want it if you are dealing with possible duplicates. I do know that the MapSet version does a sort before hand so that it doesn't have to walk the entire list. I don't think the list version does that.